contracts/ens/ENSSubdomainRegistrar.sol

```diff
@@ -0,0 +1,82 @@
+pragma solidity 0.4.18;
+
+import "./AbstractENS.sol";
+import "./PublicResolver.sol";
+import "./ENSConstants.sol";
+
+import "../apps/AragonApp.sol";
+import "../common/Initializable.sol";
+
+
+contract ENSSubdomainRegistrar is AragonApp, Initializable, ENSConstants {
+    bytes32 constant public CREATE_NAME_ROLE = bytes32(1);
+    bytes32 constant public DELETE_NAME_ROLE = bytes32(2);
+    bytes32 constant public POINT_ROOTNODE_ROLE = bytes32(3);
+    // TODO: Add claim root name
+
+    AbstractENS public ens;
+    bytes32 public rootNode;
+
+    event NewName(bytes32 indexed node, bytes32 indexed label);
+    event DeleteName(bytes32 indexed node, bytes32 indexed label);
+
+    function initialize(AbstractENS _ens, bytes32 _rootNode) onlyInit public {
+        initialized();
+
+        // We need ownership to create subnodes
+        require(_ens.owner(_rootNode) == address(this));
+
+        ens = _ens;
+        rootNode = _rootNode;
+    }
+
+    function createName(bytes32 _label, address _owner) auth(CREATE_NAME_ROLE) external returns (bytes32 node) {
+        return _createName(_label, _owner);
+    }
+
+    function createNameAndPoint(bytes32 _label, address _target) auth(CREATE_NAME_ROLE) external returns (bytes32 node) {
+        node = _createName(_label, this);
+        _pointToResolverAndResolve(node, _target);
+    }
+
+    function deleteName(bytes32 _label) auth(DELETE_NAME_ROLE) external {
+        bytes32 node = keccak256(rootNode, _label);
+
+        address currentOwner = ens.owner(node);
+
+        require(currentOwner != address(0)); // fail if deleting unset name
+
+        if (currentOwner != address(this)) // needs to reclaim ownership so it can set resolver
+            ens.setSubnodeOwner(rootNode, _label, this);
+
+        ens.setResolver(node, address(0)); // remove resolver so it ends resolving
+        ens.setOwner(node, address(0));
```

Comment:
Do you think we should also reset the TTL, just in case?

Reply:
I dont know any app that uses the TTL, I would save the extra call until that becomes relevant if ever.